x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -729,21 +730,27 @@ public Literal visitDecimalLiteral(DecimalLiteralContext ctx) {
     public Literal visitIntegerLiteral(IntegerLiteralContext ctx) {
         Tuple<Source, String> tuple = withMinus(ctx);
 
-        long value;
+        Number value;
         try {
-            value = Long.valueOf(StringUtils.parseLong(tuple.v2()));
+            value = StringUtils.parseIntegral(tuple.v2());
```

Comment:
Could `parseIntegral` also directly return the best fitting `Number` type? Right now the downsizing from `BigInteger` to `Long` happens in `parseIntegral` and from `Long` to `Integer` here and in  the EQL `ExpressionBuilder`

Reply:
Good point, fixed, thanks.